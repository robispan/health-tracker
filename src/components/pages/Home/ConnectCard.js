import React from 'react';

const Card = ({ imageUrl, connected, onConnect }) => {
  const classList = ['connect-card-link'];
  if (connected) classList.push('connected');

  return (
    <div className='connect-card'>
      <div
        className='connect-card-image'
        style={{ backgroundImage: `url(${imageUrl})` }}
      ></div>
      <a
        href='#'
        onClick={connected ? null : onConnect}
        className={classList.join(' ')}
      >
        {connected ? 'Connected' : 'Connect'}
      </a>
    </div>
  );
};

export default Card;
