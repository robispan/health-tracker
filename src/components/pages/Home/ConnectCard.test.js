import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import StravaImg from '../../assets/images/strava.jpg';
import ConnectCard from './ConnectCard';

afterEach(cleanup);

it('renders', () => {
  const { asFragment } = render(
    <ConnectCard imageUrl={StravaImg} connected={false} />
  );
  expect(asFragment()).toMatchSnapshot();
});

it('inserts text to <a> tag and adds the right css class', () => {
  const { getByText } = render(
    <ConnectCard imageUrl={StravaImg} connected={true} />
  );
  expect(getByText('Disconnect')).toHaveClass('connected');
});
