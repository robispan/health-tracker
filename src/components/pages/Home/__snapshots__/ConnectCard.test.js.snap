// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="connect-card"
  >
    <div
      class="connect-card-image"
      style="background-image: url(strava.jpg);"
    />
    <a
      class="connect-card-button"
      href="#"
    >
      Connect
    </a>
  </div>
</DocumentFragment>
`;
